--- a/config.mk	2016-11-13 19:48:58.000000000 +0300
+++ b/config.mk	2016-11-13 19:50:17.000000000 +0300
@@ -26,28 +26,28 @@
 # CFLAGS                                                                      #
 ###############################################################################
 
-CFLAGS_OPT := \
+#CFLAGS_OPT := \
   -Os \
   -flto \
   -fstrict-aliasing
 
-CFLAGS_WARN := \
+#CFLAGS_WARN := \
   -Weverything \
   -Werror \
   -Wno-unknown-warning-option \
   -Wno-reserved-id-macro \
   -pedantic
 
-CFLAGS_DIAG := \
+#CFLAGS_DIAG := \
   -fmessage-length=152 \
   -fdiagnostics-show-note-include-stack \
   -fmacro-backtrace-limit=0 \
   -fcolor-diagnostics
 
-CFLAGS_DBG := \
+#CFLAGS_DBG := \
   -g
 
-CFLAGS := \
+#CFLAGS := \
   -arch x86_64 \
   -x c \
   -std=c11 \
