--- a/lib/Makefile.darwin-lib	2016-11-19 03:48:13.000000000 +0300
+++ b/lib/Makefile.darwin-lib	2016-11-19 04:25:53.000000000 +0300
@@ -1,5 +1,5 @@
 #
-# This is a Makefile stub which handles the creation of Darwin BSD shared
+# This is a Makefile stub which handles the creation of Darwin shared
 # libraries.
 #
 # In order to use this stub, the following makefile variables must be defined.
@@ -8,44 +8,61 @@
 # BSDLIB_IMAGE = libce
 # BSDLIB_MYDIR = et
 # BSDLIB_INSTALL_DIR = $(SHLIBDIR)
-#
+# ELF_OTHER_LIBS = -lc
 
-all:: image
+all:: pic image
 
 real-subdirs:: Makefile
 	$(E) "	MKDIR pic"
 	$(Q) mkdir -p pic
 
-BSD_LIB = $(BSDLIB_IMAGE).$(BSDLIB_VERSION).dylib
-BSDLIB_PIC_FLAG = -fPIC
+ELF_LIB = $(ELF_IMAGE).$(ELF_VERSION).dylib
+ELF_SONAME = $(ELF_IMAGE).$(ELF_SO_VERSION).dylib
 
-image:		$(BSD_LIB)
+image:		$(ELF_LIB)
 
-$(BSD_LIB): $(OBJS)
-	$(E) "	GEN_BSD_SOLIB $(BSD_LIB)"
-	$(Q) (cd pic; $(CC) -dynamiclib -compatibility_version 1.0 -current_version $(BSDLIB_VERSION) \
-		-flat_namespace -undefined warning -o $(BSD_LIB) $(OBJS))
-	$(Q) $(MV) pic/$(BSD_LIB) .
-	$(Q) $(RM) -f ../$(BSD_LIB)
-	$(Q) (cd ..; $(LN) $(LINK_BUILD_FLAGS) \
-		`echo $(my_dir) | sed -e 's;lib/;;'`/$(BSD_LIB) $(BSD_LIB))
-	$(Q) (cd ..; $(LN) $(LINK_BUILD_FLAGS) $(BSD_LIB) $(BSDLIB_IMAGE).dylib)
-
-install-shlibs install:: $(BSD_LIB)
-	$(E) "	INSTALL_PROGRAM $(BSDLIB_INSTALL_DIR)/$(BSD_LIB)"
-	$(Q) $(INSTALL_PROGRAM) $(BSD_LIB) \
-		$(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSD_LIB)
-	-$(Q) $(LDCONFIG)
+$(ELF_LIB): $(OBJS)
+	@echo "	GEN_ELF_SOLIB $(ELF_LIB)"
+	@(cd pic; $(CC) -dynamiclib -compatibility_version 1.0 -flat_namespace -undefined warning -o $(ELF_LIB) \
+		-Wl,-install_name,$(libdir)/$(ELF_SONAME) $(OBJS) $(ELF_OTHER_LIBS))
+	@$(MV) pic/$(ELF_LIB) .
+	@$(RM) -f ../$(ELF_LIB) ../$(ELF_IMAGE).dylib ../$(ELF_SONAME)
+	@$(LN) $(ELF_LIB) ../$(ELF_LIB)
+	@$(LN) ../$(ELF_LIB) ../$(ELF_IMAGE).dylib
+	@$(LN) ../$(ELF_LIB) ../$(ELF_SONAME)
+
+installdirs-elf-lib::
+	@echo "	MKINSTALLDIRS $(ELF_INSTALL_DIR) $(libdir)"
+	@$(MKINSTALLDIRS) $(DESTDIR)$(ELF_INSTALL_DIR) \
+		$(DESTDIR)$(libdir)
+
+installdirs:: installdirs-elf-lib
+
+install-shlibs install:: $(ELF_LIB) installdirs-elf-lib
+	@echo "	INSTALL-ELF-LIB $(ELF_INSTALL_DIR)/$(ELF_LIB)"
+	@$(INSTALL_PROGRAM) $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_LIB)
+	@echo "	SYMLINK $(ELF_INSTALL_DIR)/$(ELF_SONAME)"
+	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
+	@echo "	SYMLINK $(libdir)/$(ELF_IMAGE).dylib"
+	@$(LN_S) -f $(ELF_SONAME) \
+		$(DESTDIR)$(libdir)/$(ELF_IMAGE).dylib
 
 install-strip: install
+	@echo "	STRIP-LIB $(ELF_INSTALL_DIR)/$(ELF_LIB)"
+	@$(STRIP) --strip-unneeded --remove-section=.comment \
+		--remove-section=.note $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_LIB)
 
 install-shlibs-strip: install-shlibs
+	@echo "	STRIP-LIB $(ELF_INSTALL_DIR)/$(ELF_LIB)"
+	@$(STRIP) --strip-unneeded --remove-section=.comment \
+		--remove-section=.note $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_LIB)
 
 uninstall-shlibs uninstall::
-	$(RM) -f $(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSD_LIB)
+	$(RM) -f $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_LIB) \
+		$(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME) \
+		$(DESTDIR)$(libdir)/$(ELF_IMAGE).dylib
 
 clean::
 	$(RM) -rf pic
-	$(RM) -f $(BSD_LIB)
-	$(RM) -f ../$(BSD_LIB)
-	$(RM) -f ../$(BSDLIB_IMAGE).dylib
+	$(RM) -f $(ELF_LIB)
+	$(RM) -f ../$(ELF_LIB) ../$(ELF_IMAGE).dylib ../$(ELF_SONAME)
